src/olympia/addons/tasks.py

```diff
@@ -413,3 +415,61 @@ def populate_e10s_feature_compatibility(ids, **kwargs):
                 AddonFeatureCompatibility.objects.get_or_create(addon=addon)[0]
             )
             feature_compatibility.update(e10s=parsed['e10s_compatibility'])
+
+
+@task
+@write
+def migrate_preliminary_to_full(ids, **kwargs):
+    log.info(
+        '[%s@%s] Removing preliminary status from all addons starting w/ id: '
+        '%s...' % (len(ids), migrate_preliminary_to_full.rate_limit, ids[0]))
+    addons = Addon.unfiltered.filter(pk__in=ids)
+    ids_to_email = []
+
+    for addon in addons:
+        # Add-on is preliminary reviewed, so make it full and experimental.
+        if addon.status == amo.STATUS_LITE:
+            File.objects.filter(
+                version__addon=addon,
+                status=amo.STATUS_LITE).update(status=amo.STATUS_PUBLIC)
+            addon.update(status=amo.STATUS_PUBLIC, is_experimental=True)
+            add_note = True
+        # Add-on is waiting for preliminary approval, so make it nominated and
+        # experimental.
+        elif addon.status == amo.STATUS_UNREVIEWED:
+            addon.update(status=amo.STATUS_NOMINATED, is_experimental=True)
+            add_note = True
+        # Add-on is preliminary approved and nominated for full, so make it
+        # nominated.
+        elif addon.status == amo.STATUS_LITE_AND_NOMINATED:
+            # We need a file to approve so unapprove it. Rest will be disabled.
```

Comment:
I think in this case it would be better to auto-approve the nomination. So, treat like `STATUS_LITE`, but without the experimental flag. If there's a file pending review, keep it pending.